eduNEXT/eox-audit-model
# feat: make eox-audit-model comptatible with nutmeg

# Migrate Eox-audit-model to nutmeg compatibility 

## Description
This PR makes follow changes:

- Drop Django 2.2 and python 3.5 compatibilities.
- Update requirements according to nutmeg release.
- Remove pylint unused disable rules and add one (`disable=used-before-assignment`). 
- Add `DEFAULT_AUTO_FIELD` to generate primary keys. [Reference doc](https://docs.djangoproject.com/en/3.2/releases/3.2/#customizing-type-of-auto-created-primary-keys)
- Remove `get_ip try-except`  due to [django-ipware ](https://github.com/openedx/edx-platform/blob/open-release/nutmeg.master/requirements/edx/base.txt#L295) update now use `get_client_ip` [refentence](https://github.com/un33k/django-ipware/tree/v4.0.2#how-to-use).

## Testing instructions

### New environment

- Use stack builder with tutor 14.0.5 
```YML

STRAIN_NAME: nutmeg
STRAIN_TUTOR_VERSION: v14.0.5
DEV_PROJECT_NAME: nutmeg_dev
LOCAL_PROJECT_NAME: nutmeg_local
DOCKER_IMAGE_OPENEDX: docker.io/ednxops/distro-edunext-edxapp:nuez-audit
STRAIN_TUTOR_PLUGINS:
  - REPO: tutor-contrib-edunext-distro
    VERSION: v3.0.0-rc1
    EGG: tutor-contrib-edunext-distro==v3.0.0-rc1
    PACKAGE_NAME: distro
    PROTOCOL: ssh
CMS_HOST: studio.nutmeg.edunext.link
LMS_HOST: lms.nutmeg.edunext.link
EOX_AUDIT_MODEL_DPKG:
    index: git
    name: eox-audit-model
    repo: eox-audit-model
    version: dcoa/nutmeg-migration-DS-252
    domain: github.com
    protocol: https
    path: eduNEXT
    private: False
    variables:
      development: {}
      production: {}
DISTRO_EOX_CORE_DPKG:
  index: git
  name: eox-core
  repo: eox-core
  private: false
  version: mfmz/compatible-with-nutmeg
  domain: github.com
  protocol: https
  path: eduNEXT
  variables: 
    development:
      EOX_CORE_USERS_BACKEND: eox_core.edxapp_wrapper.backends.users_m_v1
      EOX_CORE_ENROLLMENT_BACKEND: eox_core.edxapp_wrapper.backends.enrollment_l_v1
      EOX_CORE_PRE_ENROLLMENT_BACKEND: eox_core.edxapp_wrapper.backends.pre_enrollment_l_v1
    production:
        EOX_CORE_USERS_BACKEND: eox_core.edxapp_wrapper.backends.users_m_v1
        EOX_CORE_ENROLLMENT_BACKEND: eox_core.edxapp_wrapper.backends.enrollment_l_v1
        EOX_CORE_PRE_ENROLLMENT_BACKEND: eox_core.edxapp_wrapper.backends.pre_enrollment_l_v1
```
- Run `stack strain local configure` and `stack strain local init`
- Create superuser `tutor local createuser --staff --superuser admin admin@edunext.co`

### Already have a tutor environment 
- Copy the strain describe in this pr and  run `stack strain save` 
- Pull docker image with `stack strain local configure --step update_images` and `stack strain local start`
- Open a new terminal and inside the container run migrations 
```bash
tutor local run lms bash
       # lms bash
       python manage.py lms migrate
```

### Test eox-audit-model test cases
- Create the eox-core user and application, follow instructions in the [eox-core test case document - 2.6. TESTING APIs WITH A USER WITHOUT STAFF PRIVILEGES](https://docs.google.com/document/d/1gaQyHBGDDkvy_NsFw07MRV6uI2R49CeWCDHzTVi_TiI/edit)
- Use the eox-core API collection [find here](https://edunext.atlassian.net/jira/software/c/projects/DS/boards/40?modal=detail&selectedIssue=DS-186) if you don't know how to use it follow [this instructions](https://docs.google.com/document/d/16ERm4roHyrL5F8kHqqtABr9J4jtPC-kFH8TnQ14pa90/edit)
- Create an environment with token and domain, and update your token with an authorization request.
![eox-core-api-postman](https://user-images.githubusercontent.com/66016493/192907000-1e4b1ec5-68ad-4e25-8967-a5a11146d730.png)

- Test eox-audit-model ([test cases](https://docs.google.com/document/d/1v6mxFSf1k7JrxU36Ok9uVDFBhtxVuGmhFLtxtbIsPFs/edit#)) create a user and check the success audit register, run the same create user request again to check the fail audit register.
## Checklist for Merge
